### PR TITLE
197 Fix TCR bug

### LINKED ISSUE #197: Server 500 when tries to add a TCR rule to Group
```
15:08:20.488 [http-nio-8080-exec-1] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.NullPointerException] with root cause
java.lang.NullPointerException: null
	at innohack.gem.entity.rule.rules.TikaContentRegex.check(TikaContentRegex.java:37)
	at innohack.gem.service.MatchService.checkMatching(MatchService.java:100)
	at innohack.gem.service.MatchService.onUpdateGroupRule(MatchService.java:196)
	at innohack.gem.service.MatchService.onUpdateEvent(MatchService.java:49)
	at innohack.gem.service.GroupService.saveGroup(GroupService.java:72)
	at innohack.gem.web.GroupController.saveGroup(GroupController.java:97)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:106)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:879)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:793)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
```

### PR DESCRIPTION
**Pull request by yclim**
close #197

### CODE REVIEW

File: src/main/java/innohack/gem/entity/rule/rules/TikaContentRegex.java [hunk 1]
```diff
@@ -51,9 +51,9 @@ private boolean checkTikaContent(TikaFeature abs) {
       Pattern pattern = Pattern.compile(regexString, Pattern.CASE_INSENSITIVE);
       Matcher matcher = pattern.matcher(tikaContent);
 
-      while (matcher.find()) {
-        String matchString = matcher.group(1);
-        LOGGER.info("The  match string is {}", matchString);
+      if (matcher.find()) {
+        LOGGER.info(
+            "The match string is {}", tikaContent.substring(matcher.start(), matcher.end()));
```
Comment (onegang) [inline src/main/java/innohack/gem/entity/rule/rules/TikaContentRegex.java:56]: we can also do a group(0) which gives the match

File: src/test/java/innohack/gem/entity/rule/rules/TikaContentRegexTest.java [hunk 2]
```diff
@@ -17,6 +17,18 @@ public void testValidCheckTikaContent() throws Exception {
     assertTrue(contentRegexCheck.check(pdfFile));
   }
 
+  @Test
+  public void testValidCheckWithoutCaptureGroup() throws Exception {
+
+    GEMFile pdfFile = new GEMFile("story_0.pdf", "src/test/resources/extract");
+    pdfFile.extract();
+    TikaContentRegex contentRegexCheck = new TikaContentRegex("[white|black]");
```
Comment (onegang) [inline src/test/java/innohack/gem/entity/rule/rules/TikaContentRegexTest.java:25]: This test for the presence of any letters `w, h, i, t, e, |, b, l, a, c, k`. Is that the intention? If not, use a word boundary match?

**Comment by onegang** [comment]
I will insert another patch for the `TikaContentExtractor` which assumes group usage for regex too.

**Comment by onegang** [comment]
> TikaContentExtractor

Ok, I have inserted my patch into this PR

File: src/test/java/innohack/gem/entity/rule/rules/TikaContentRegexTest.java (hunk 2 above)
Comment (yclim) [reply src/test/java/innohack/gem/entity/rule/rules/TikaContentRegexTest.java:25]: oh thats a mistake. i actually want word boundary or non-capturing group ie `(?:white|back)`

**Comment by yclim** [comment]
i think capture group is fine for regex extractor because we need to extract multiple parts of the string. In the UI, i use num of capture groups to check if it matches the num of columns specified for the extraction.

**Comment by yclim** [comment]
inserted the fix

**Pull request merged**